dxp/demystifying_aui_script_and_liferay-frontend_component_tags.md

```diff
@@ -0,0 +1,114 @@
+# Demystifying `aui:script` and `liferay-frontend:component` tags
+
+On JSP context you may notice a huge number of `aui:script` and `liferay-frontend:component` tags.
+
+## `aui:script` tag
+
+### What it does:
+
+This taglib allow users to use and require another JS module/resources inside a JSP.
+
+This taglib have three possible use cases:
+
+1. Pure JS on the JSP
+2. Import a AUI/YUI module
+3. Import a ES6 module
+
+#### Pure JS on the JSP
+
+By just placing JavaScript code inside an `aui:script` tag, you can take advantage of optimizing scripts and place them to be made available in the final compressed JavaScript bundle, see [combo for more details](https: / /github.com/liferay/liferay-frontend-guidelines/blob/45bad0334cb467e45e68aee322a2a4bc0b6979d3/dxp/javaScript_minification.md#comboservlet). It is worth mentioning that the same rule applies for the following use cases.
```

Comment:
Weird spacing around the ` combo for more details` link. It dos not seem to generate a valid url.

Reply:
I think this is a bit confusing because it's not clear how `aui:script` is related to ComboServlet...

Reply:
> I think this is a bit confusing because it's not clear how aui:script is related to ComboServlet...

Does it looks fine?

By just placing JavaScript code inside an `aui:script` tag, you can take advantage of optimizing scripts and place them to be made available in the final compressed JavaScript bundle. It can be achieved when appending the script to the [`AUI_SCRIPT_DATA` WebKey](https://github.com/liferay/liferay-portal/blob/67b569099146a4b999e2fad7d7d1a9794a337f0b/portal-kernel/src/com/liferay/portal/kernel/util/WebKeys.java#L55) and all these scripts will be applied at the [bottom of the page](https://github.com/liferay/liferay-portal/blob/e3ffac158e0ec5acc5c67069fbd7ba688d3c78d4/portal-web/docroot/html/common/themes/bottom.jsp#L44). Finally the [Aggregate Filter will take care](https://github.com/liferay/liferay-frontend-guidelines/blob/master/dxp/resource_serving.md#aggregate-filter), of creating the final bundle with all the JavaScript on the page. It is worth mentioning that the same rule applies for all the following use cases.

Reply:
That seems better, but I am not confident enough about my knowledge of the system to say whether it is right or not. @izaera: can you confirm that this is factually correct?

Reply:
> Finally the Aggregate Filter will take care, of creating the final bundle with all the JavaScript on the page. It is worth mentioning that the same rule applies to all the following use cases.

I can't find any evidence of this part being true. In fact, it looks like [it's this guy](https://github.com/liferay/liferay-portal/blob/ebed38f329f4db6bfae067c46a26da3397597f2b/portal-impl/src/com/liferay/portlet/ScriptDataPortletFilter.java#L70) who writes it...